protected/humhub/modules/content/jobs/ReindexUserContent.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * @link https://www.humhub.org/
+ * @copyright Copyright (c) HumHub GmbH & Co. KG
+ * @license https://www.humhub.com/licences
+ */
+
+namespace humhub\modules\content\jobs;
+
+use humhub\modules\content\models\Content;
+use humhub\modules\content\Module;
+use humhub\modules\content\search\driver\AbstractDriver;
+use humhub\modules\content\services\SearchJobService;
+use humhub\modules\queue\interfaces\ExclusiveJobInterface;
+use humhub\modules\queue\LongRunningActiveJob;
+use humhub\modules\user\models\User;
+use Yii;
+
+class ReindexUserContent extends LongRunningActiveJob implements ExclusiveJobInterface
+{
+    public $userId;
+
+    /**
+     * @inhertidoc
+     */
+    public function getExclusiveJobId()
+    {
+        return 'content-search.reindex-user.' . $this->userId;
+    }
+
+    /**
+     * @inhertidoc
+     */
+    public function run()
+    {
+        return $this->getService()->run(function () {
+            $user = User::findOne(['id' => $this->userId]);
+            if (!$user) {
+                return;
+            }
+
+            $contents = Content::find()
+                ->where(['created_by' => $user->id]);
+
+            foreach ($contents->each() as $content) {
+                if ($user->status === User::STATUS_ENABLED) {
+                    $this->getSearchDriver()->update($content);
```

Comment:
Maybe it's better here to use `ContentSearchService::update()`?

Reply:
@luke- Commit https://github.com/humhub/humhub/pull/7153/commits/a3cf6251275455397e8bcbb27f58a6c4c7683010.